crates/flight_datafusion/src/lib.rs

```diff
@@ -270,16 +271,20 @@ impl ExecutionPlan for FlightExec {
             Err(error) => return Err(error),
         };
 
-        Ok(Box::pin(StreamConverter::new(
-            self.client.clone(),
-            sql.as_str(),
+        let stream_adapter = RecordBatchStreamAdapter::new(
             self.schema(),
-        )))
+            query_to_stream(self.client.clone(), sql.as_str()),
+        );
+
+        Ok(Box::pin(stream_adapter))
     }
 }
 
 #[allow(clippy::needless_pass_by_value)]
-fn to_stream(client: FlightClient, sql: &str) -> impl Stream<Item = Result<RecordBatch>> {
+fn query_to_stream(
+    client: FlightClient,
```

Comment:
is this expected  that the client is then owned by the function? I would have expected the statement at line 288 to be outside, i.e. at invocation site

Reply:
Actually it is at invocation site, maybe you can only have a mut Flightclient at line 285?

Reply:
Yeah, I made it a `mut FlightClient` - thanks for catching this.